application/common/models/User.php

```diff
@@ -24,12 +24,20 @@ class User extends UserBase
     const SCENARIO_AUTHENTICATE    = 'authenticate';
     const SCENARIO_INVITE          = 'invite';
 
```

Comment:
would encapsulating these states in a class make the code neater?  It might also give you the opportunity to extract some of the state logic, e.g., determining current state, transitioning to another state, etc., into a more cohesive implementation.